test/integration/volumescheduling/util.go

```diff
@@ -102,56 +103,64 @@ func initTestMaster(t *testing.T, nsPrefix string, admission admission.Interface
+	podInformer := testContext.informerFactory.Core().V1().Pods()
 
-	context.schedulerConfigArgs = createConfiguratorArgsWithPodInformer(
-		context.clientSet, podInformer, context.informerFactory, schedulerframework.NewRegistry(), nil,
-		[]schedulerconfig.PluginConfig{}, context.stopCh)
-	configFactory := factory.NewConfigFactory(context.schedulerConfigArgs)
+	testContext.schedulerConfigArgs = createConfiguratorArgsWithPodInformer(
+		testContext.clientSet, podInformer, testContext.informerFactory, schedulerframework.NewRegistry(), nil,
+		[]schedulerconfig.PluginConfig{}, testContext.stopCh)
+	configFactory := factory.NewConfigFactory(testContext.schedulerConfigArgs)
 
 	var err error
-	context.schedulerConfig, err = configFactory.Create()
+	testContext.schedulerConfig, err = configFactory.Create()
 	if err != nil {
 		t.Fatalf("Couldn't create scheduler config: %v", err)
 	}
 
 	// set DisablePreemption option
-	context.schedulerConfig.DisablePreemption = false
+	testContext.schedulerConfig.DisablePreemption = false
 	eventBroadcaster := events.NewBroadcaster(&events.EventSinkImpl{
-		Interface: context.clientSet.EventsV1beta1().Events(""),
+		Interface: testContext.clientSet.EventsV1beta1().Events(""),
 	})
-	context.schedulerConfig.Recorder = eventBroadcaster.NewRecorder(
+	testContext.schedulerConfig.Recorder = eventBroadcaster.NewRecorder(
 		legacyscheme.Scheme,
 		v1.DefaultSchedulerName,
 	)
 
-	context.scheduler = scheduler.NewFromConfig(context.schedulerConfig)
+	testContext.scheduler = scheduler.NewFromConfig(testContext.schedulerConfig)
 
-	scheduler.AddAllEventHandlers(context.scheduler,
+	scheduler.AddAllEventHandlers(testContext.scheduler,
 		v1.DefaultSchedulerName,
-		context.informerFactory.Core().V1().Nodes(),
+		testContext.informerFactory.Core().V1().Nodes(),
 		podInformer,
-		context.informerFactory.Core().V1().PersistentVolumes(),
-		context.informerFactory.Core().V1().PersistentVolumeClaims(),
-		context.informerFactory.Core().V1().Services(),
-		context.informerFactory.Storage().V1().StorageClasses(),
-		context.informerFactory.Storage().V1beta1().CSINodes(),
+		testContext.informerFactory.Core().V1().PersistentVolumes(),
+		testContext.informerFactory.Core().V1().PersistentVolumeClaims(),
+		testContext.informerFactory.Core().V1().Services(),
+		testContext.informerFactory.Storage().V1().StorageClasses(),
+		testContext.informerFactory.Storage().V1beta1().CSINodes(),
 	)
 
 	stopCh := make(chan struct{})
 	eventBroadcaster.StartRecordingToSink(stopCh)
 
-	context.informerFactory.Start(context.schedulerConfig.StopEverything)
-	context.informerFactory.WaitForCacheSync(context.schedulerConfig.StopEverything)
+	testContext.informerFactory.Start(testContext.schedulerConfig.StopEverything)
+	testContext.informerFactory.WaitForCacheSync(testContext.schedulerConfig.StopEverything)
 
-	context.scheduler.Run()
-	return context
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
The same as test/integration/util/util.go. We should tweak testContext to embed a context.Context.